2024-07-11 12:34:48.059404: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:34:48.059595: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:34:56,419 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:34:56,514 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:34:56,514 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:34:56,514 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:34:57,812 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:34:57,812 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:57,834 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.47333333333333333}
INFO flwr 2024-07-11 12:34:57,834 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:34:58,112 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:34:58,135 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:34:58,137 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,145 | server.py:131 | fit progress: (1, 0.4086602151209276, {'accuracy': 0.8366666666666667}, 0.3111712999999998)
DEBUG flwr 2024-07-11 12:34:58,145 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:34:58,167 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:34:58,167 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:34:58,168 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:34:58,191 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,204 | server.py:131 | fit progress: (2, 0.4019301337475059, {'accuracy': 0.84}, 0.3704799000000012)
DEBUG flwr 2024-07-11 12:34:58,205 | server.py:178 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:34:58,227 | server.py:192 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:34:58,228 | server.py:227 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:34:58,256 | server.py:241 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,268 | server.py:131 | fit progress: (3, 0.4076533665861525, {'accuracy': 0.8433333333333334}, 0.43325570000000013)
DEBUG flwr 2024-07-11 12:34:58,268 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:34:58,288 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:34:58,289 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:34:58,319 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,331 | server.py:131 | fit progress: (4, 0.40738180322294276, {'accuracy': 0.8433333333333334}, 0.49598670000000133)
DEBUG flwr 2024-07-11 12:34:58,331 | server.py:178 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:34:58,350 | server.py:192 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:34:58,351 | server.py:227 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:34:58,378 | server.py:241 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,389 | server.py:131 | fit progress: (5, 0.40760427406808136, {'accuracy': 0.8433333333333334}, 0.5555716000000004)
DEBUG flwr 2024-07-11 12:34:58,390 | server.py:178 | evaluate_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:34:58,411 | server.py:192 | evaluate_round 5 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:34:58,411 | server.py:227 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:34:58,443 | server.py:241 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,455 | server.py:131 | fit progress: (6, 0.4073156799601375, {'accuracy': 0.8366666666666667}, 0.6205246000000013)
DEBUG flwr 2024-07-11 12:34:58,455 | server.py:178 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:34:58,480 | server.py:192 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:34:58,480 | server.py:227 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:34:58,509 | server.py:241 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,520 | server.py:131 | fit progress: (7, 0.40732700022710927, {'accuracy': 0.8366666666666667}, 0.6861467000000001)
DEBUG flwr 2024-07-11 12:34:58,521 | server.py:178 | evaluate_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:34:58,551 | server.py:192 | evaluate_round 7 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:34:58,552 | server.py:227 | fit_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:34:58,584 | server.py:241 | fit_round 8 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,595 | server.py:131 | fit progress: (8, 0.4073270193132361, {'accuracy': 0.8366666666666667}, 0.7617440000000002)
DEBUG flwr 2024-07-11 12:34:58,596 | server.py:178 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:34:58,627 | server.py:192 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:34:58,627 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:34:58,667 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,679 | server.py:131 | fit progress: (9, 0.4101844907153074, {'accuracy': 0.83}, 0.8448056000000008)
DEBUG flwr 2024-07-11 12:34:58,679 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:34:58,716 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:34:58,716 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:34:58,757 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:34:58,770 | server.py:131 | fit progress: (10, 0.41018373605343056, {'accuracy': 0.83}, 0.9356274000000013)
DEBUG flwr 2024-07-11 12:34:58,770 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:34:58,799 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:34:58,800 | server.py:153 | FL finished in 0.9651562000000009
INFO flwr 2024-07-11 12:34:58,800 | app.py:225 | app_fit: losses_distributed [(1, 0.3981585595756769), (2, 0.4165284663438797), (3, 0.41574174066384634), (4, 0.415728947520256), (5, 0.4157359460989634), (6, 0.4117524517433984), (7, 0.4117522495133536), (8, 0.41503234803676603), (9, 0.4149142499480929), (10, 0.41491421588829586)]
INFO flwr 2024-07-11 12:34:58,800 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:34:58,800 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:34:58,800 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.4086602151209276), (2, 0.4019301337475059), (3, 0.4076533665861525), (4, 0.40738180322294276), (5, 0.40760427406808136), (6, 0.4073156799601375), (7, 0.40732700022710927), (8, 0.4073270193132361), (9, 0.4101844907153074), (10, 0.41018373605343056)]
INFO flwr 2024-07-11 12:34:58,801 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.47333333333333333), (1, 0.8366666666666667), (2, 0.84), (3, 0.8433333333333334), (4, 0.8433333333333334), (5, 0.8433333333333334), (6, 0.8366666666666667), (7, 0.8366666666666667), (8, 0.8366666666666667), (9, 0.83), (10, 0.83)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.96502125,  1.0450829 ,  0.83114467,  0.38674169,  0.24218687,
         0.00346211, -0.01921992, -0.06451128]]), array([0.09649747])]
[array([[-1.96502125,  1.0450829 ,  0.83114467,  0.38674169,  0.24218687,
         0.00346211, -0.01921992, -0.06451128]]), array([0.09649747])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.94752569,  1.08431438,  0.82761415,  0.3906515 ,  0.26074774,
         0.05019113, -0.03307257, -0.09734294]]), array([-0.03874793])]
[array([[-1.94752569,  1.08431438,  0.82761415,  0.3906515 ,  0.26074774,
         0.05019113, -0.03307257, -0.09734294]]), array([-0.03874793])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89661454,  0.96808057,  0.80087938,  0.3607401 ,  0.23372345,
         0.0339688 ,  0.00384686, -0.03740221]]), array([0.01369898])]
[array([[-1.89661454,  0.96808057,  0.80087938,  0.3607401 ,  0.23372345,
         0.0339688 ,  0.00384686, -0.03740221]]), array([0.01369898])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.89978461,  0.96841908,  0.80035657,  0.36079796,  0.233064  ,
         0.03555654,  0.00288608, -0.04159827]]), array([0.01953984])]
[array([[-1.89978461,  0.96841908,  0.80035657,  0.36079796,  0.233064  ,
         0.03555654,  0.00288608, -0.04159827]]), array([0.01953984])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.8983959 ,  0.96753517,  0.80041132,  0.36027827,  0.23296969,
         0.03422063,  0.00342202, -0.0395023 ]]), array([0.02004927])]
[array([[-1.8983959 ,  0.96753517,  0.80041132,  0.36027827,  0.23296969,
         0.03422063,  0.00342202, -0.0395023 ]]), array([0.02004927])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.9589186 ,  0.9655012 ,  0.82259566,  0.31649827,  0.1977186 ,
         0.04859996,  0.03758327, -0.04197047]]), array([0.05102573])]
[array([[-1.9589186 ,  0.9655012 ,  0.82259566,  0.31649827,  0.1977186 ,
         0.04859996,  0.03758327, -0.04197047]]), array([0.05102573])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.95926184,  0.96557855,  0.82241744,  0.3163234 ,  0.19754281,
         0.04845029,  0.03752771, -0.04191245]]), array([0.05200404])]
[array([[-1.95926184,  0.96557855,  0.82241744,  0.3163234 ,  0.19754281,
         0.04845029,  0.03752771, -0.04191245]]), array([0.05200404])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.95926104,  0.96558304,  0.82240481,  0.31632346,  0.19753526,
         0.04845112,  0.03752627, -0.04190933]]), array([0.05201701])]
[array([[-1.95926104,  0.96558304,  0.82240481,  0.31632346,  0.19753526,
         0.04845112,  0.03752627, -0.04190933]]), array([0.05201701])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.94091433,  0.93747169,  0.80679374,  0.32163781,  0.1834782 ,
         0.01510495,  0.03405535, -0.04733409]]), array([0.14855731])]
[array([[-1.94091433,  0.93747169,  0.80679374,  0.32163781,  0.1834782 ,
         0.01510495,  0.03405535, -0.04733409]]), array([0.14855731])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.94093648,  0.93744875,  0.80685139,  0.32168351,  0.18353195,
         0.01507338,  0.03398903, -0.04735351]]), array([0.14857135])]
[array([[-1.94093648,  0.93744875,  0.80685139,  0.32168351,  0.18353195,
         0.01507338,  0.03398903, -0.04735351]]), array([0.14857135])]
>>>>>>>>>>>>>run time:
0.640625s
